File: test/lit/zig/float_types.zig
// RUN: %cot emit-cir %s | %FileCheck %s

pub fn add_f32(a: f32, b: f32) f32 {
    return a + b;
}

pub fn add_f64(a: f64, b: f64) f64 {
    return a + b;
}

// CHECK-LABEL: func.func @add_f32
// CHECK: cir.add %{{.*}}, %{{.*}} : f32

// CHECK-LABEL: func.func @add_f64
// CHECK: cir.add %{{.*}}, %{{.*}} : f64

File: test/lit/swift/float_types.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: Float (f32) and Double (f64) literals and arithmetic

func addFloat(a: Float, b: Float) -> Float {
    return a + b
}

func addDouble(a: Double, b: Double) -> Double {
    return a + b
}

// CHECK-LABEL: func.func @addFloat
// CHECK: cir.add %arg0, %arg1 : f32

// CHECK-LABEL: func.func @addDouble
// CHECK: cir.add %arg0, %arg1 : f64
